packages/amplify_storage_s3/lib/method_channel_storage_s3.dart

```diff
@@ -168,23 +165,14 @@ class AmplifyStorageS3MethodChannel extends AmplifyStorageS3 {
 
   StorageException _prepareExceptionForMalformedResult(
       {@required String methodName, @required String fieldName}) {
-    const exceptionType =
-        StorageExceptionType.MALFORMED_PLATFORM_CHANNEL_RESULT;
-    final Map exceptionDetails = <String, String>{
-      'operation': methodName,
-      'malformed field': '$fieldName cannot be null'
-    };
     return StorageException(
-        code: exceptionType.code,
-        message: exceptionType.message,
-        details: exceptionDetails);
+      'Error formatting platform channel result',
+      recoverySuggestion: 'Operation $methodName failed: $fieldName cannot be null.');
```

Comment:
Since this is about failing to format platform channel result, is this recovery suggestion valid for customers?

Reply:
I'm not 100% sure - this is how we had it previously.  @Ashish-Nanda ?

Reply:
This is for malformed results from over the channel, and we should still throw the exception. If we want to embed this in a specific class for library exceptions or put this as the underlying exception we can discuss the approach for what would work best.